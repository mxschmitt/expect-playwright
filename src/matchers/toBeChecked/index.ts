import { SyncExpectationResult } from "expect/build/types"
import { getElementHandle, getMessage, InputArguments } from "../utils"

const toBeChecked: jest.CustomMatcher = async function (
  ...args: InputArguments
): Promise<SyncExpectationResult> {
  try {
    const [elementHandle] = await getElementHandle(args, 0)
    const isChecked = await elementHandle.isChecked()

    return {
      pass: isChecked,
      message: () => getMessage(this, "toBeChecked", true, isChecked, ""),
    }
  } catch (err) {
    return {
      pass: false,
      message: () => err.toString(),
    }
  }
}

export default toBeChecked
