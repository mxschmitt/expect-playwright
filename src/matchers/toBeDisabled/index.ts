import { SyncExpectationResult } from "expect/build/types"
import { getElementHandle, getMessage, InputArguments } from "../utils"

const toBeDisabled: jest.CustomMatcher = async function (
  ...args: InputArguments
): Promise<SyncExpectationResult> {
  try {
    const [elementHandle] = await getElementHandle(args, 0)
    const isDisabled = await elementHandle.isDisabled()

    return {
      pass: isDisabled,
      message: () => getMessage(this, "toBeDisabled", true, isDisabled, ""),
    }
  } catch (err) {
    return {
      pass: false,
      message: () => err.toString(),
    }
  }
}

export default toBeDisabled
