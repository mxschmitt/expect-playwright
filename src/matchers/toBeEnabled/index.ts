import { SyncExpectationResult } from "expect/build/types"
import { getElementHandle, getMessage, InputArguments } from "../utils"

const toBeEnabled: jest.CustomMatcher = async function (
  ...args: InputArguments
): Promise<SyncExpectationResult> {
  try {
    const [elementHandle] = await getElementHandle(args, 0)
    const isEnabled = await elementHandle.isEnabled()

    return {
      pass: isEnabled,
      message: () => getMessage(this, "toBeEnabled", true, isEnabled, ""),
    }
  } catch (err) {
    return {
      pass: false,
      message: () => err.toString(),
    }
  }
}

export default toBeEnabled
