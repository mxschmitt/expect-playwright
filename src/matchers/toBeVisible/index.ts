import { SyncExpectationResult } from "expect/build/types"
import { getElement, getExpectedMessage, InputArguments } from "../utils"

const toBeVisible: jest.CustomMatcher = async function (
  ...args: InputArguments
): Promise<SyncExpectationResult> {
  try {
    const { elementHandle } = await getElement(...args)
    const actualVisibility = await elementHandle.isVisible()

    return {
      pass: actualVisibility,
      message: () =>
        getExpectedMessage(this, 'toBeVisible'),
    }
  } catch (err) {
    return {
      pass: false,
      message: () => err.toString(),
    }
  }
}

export default toBeVisible
