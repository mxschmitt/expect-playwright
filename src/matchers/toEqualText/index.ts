import { SyncExpectationResult } from 'expect/build/types'
import { getElementText, getMessage, InputArguments } from '../utils'

const toEqualText: jest.CustomMatcher = async function (...args: InputArguments): Promise<SyncExpectationResult>  {
  try {
    const { elementHandle, expectedValue } = await getElementText(...args)
    const actualTextContent = await elementHandle.evaluate<string | null>((el) => el.textContent)

    return {
      pass: actualTextContent === expectedValue,
      message: getMessage(this, 'toEqualText', expectedValue, actualTextContent),
    }
  } catch (err) {
    return {
      pass: false,
      message: () => err.toString()
    }
  }
}

export default toEqualText
