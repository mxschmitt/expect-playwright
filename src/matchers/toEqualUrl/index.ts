import { SyncExpectationResult } from "expect/build/types"
import { Page } from "playwright-core"
import { getMessage } from "../utils"

const toEqualUrl: jest.CustomMatcher = async function (
  page: Page,
  expectedUrl: string
): Promise<SyncExpectationResult> {
  const actualUrl = page.url()

  return {
    pass: actualUrl === expectedUrl,
    message: () => getMessage(this, "toEqualUrl", expectedUrl, actualUrl),
  }
}

export default toEqualUrl
