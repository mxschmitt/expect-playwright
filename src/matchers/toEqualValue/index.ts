import { SyncExpectationResult } from 'expect/build/types'
import { getElementText, quote, InputArguments } from '../utils'

const toEqualValue = async (...args: InputArguments): Promise<SyncExpectationResult> => {
  try {
    const { elementHandle, selector, expectedValue } = await getElementText(...args)
    /* istanbul ignore next */
    const actualTextContent = await elementHandle.evaluate((el) => (el as HTMLInputElement).value)
    if (actualTextContent?.includes(expectedValue)) {
      return {
        pass: true,
        message: () => `${quote(expectedValue)} is included in ${quote(actualTextContent)}.`
      }
    }
    return {
      pass: false,
      message: () => `${quote(expectedValue)} is not included in ${quote(actualTextContent)}${selector ? ' of ' + quote(selector) + "." : '.'}`
    }
  } catch (err) {
    return {
      pass: false,
      message: () => err.toString()
    }
  }
}

export default toEqualValue