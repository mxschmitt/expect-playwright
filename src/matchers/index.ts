import toHaveText from './toHaveText'
import toEqualText from './toEqualText'
import toHaveSelector from './toHaveSelector'
import toEqualValue from './toEqualValue'
import toHaveSelectorCount from './toHaveSelectorCount'

export default {
  toHaveText,
  toEqualText,
  toHaveSelector,
  toEqualValue,
  toHaveSelectorCount
}
