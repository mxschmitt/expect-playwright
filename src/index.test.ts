import matchers from './matchers';

import expectPlaywright from "."

describe("expect-playwright", () => {
  afterEach(async () => {
    await page.evaluate(() => document.body.innerHTML = "")
  })
  it("should apply the functions", () => {
    for (let matcher in matchers) {
      // @ts-ignore
      expect(expect(null)[matcher]).not.toBeUndefined()
    }
  })
  it("should be possible to use a not selector", async () => {
    await page.evaluate(() => {
      document.write(`<div id="foobar">zzzBarzzz</div>`)
    })
    await expect(page).not.toHaveText("This is definitely not there")
  })
  it("should be possible to use a normal selector", async () => {
    await page.evaluate(() => {
      document.write(`<div id="foobar">zzzBarzzz</div>`)
    })
    await expect(page).toHaveText("zzzBarzzz")
  })
})

describe("expectPlaywright", () => {
  afterEach(async () => {
    await page.evaluate(() => document.body.innerHTML = "")
  })
  it("should be able to handle positive cases", async () => {
    await page.evaluate(() => {
      document.write(`<div id="foobar">zzzBarzzz</div>`)
    })
    expect(await expectPlaywright(page).toHaveText("zzzBarzzz")).toBe(true)
  })
  it("should be able to handle negative cases", async () => {
    await page.evaluate(() => {
      document.write(`<div id="foobar">zzzzz</div>`)
    })
    await expect(expectPlaywright(page).toHaveText("zzzBarzzz")).rejects.toThrowErrorMatchingSnapshot()
  })
})