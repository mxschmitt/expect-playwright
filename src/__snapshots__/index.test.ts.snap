// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expectPlaywright should be able to handle negative cases 1`] = `"'zzzBarzzz' is not included in 'zzzzz'."`;
